### ISSUE TITLE
TaurusValueComboBox repeatedly invokes writeValue

### ISSUE THREAD

**Issue by mkomel** [taurus-org/taurus#651]
I have noticed that TangoValueComboBox recursively invokes writeValue (at least in TaurusQT 4.1.1). This can be observed in device servers receiving write calls in fast succession.
The problem seems to be in taurusbase.py, class TaurusBaseWritableWidget, function writeValue, lines:

    if applied:
        self.applied.emit()

If I comment those two lines out, it works ok (write operation gets invoked only once, as expected).
Essentially, the effect of self.applied.emit() here is that writeValue gets recursively invoked, until the following error occurs:
    RuntimeError: maximum recursion depth exceeded while calling a Python object.

**Comment by cpascual** [comment]
Hi, thanks for reporting...

However, it is not clear to me how to reproduce. Could you please write an example (either a code snippet or some instructions) on how to trigger this? (If possible using TangoTest DS)

**Comment by mkomel** [comment]
Hi,

I have created a small example below using the TangoTestDS. 
Problem can be seen directly in the console from where this is run (RuntimeError will be shown)

```
import sys
from taurus.external.qt import Qt
from taurus.qt.qtgui.application import TaurusApplication
from taurus.qt.qtgui.display import TaurusLabel
from taurus.qt.qtgui.input import TaurusValueComboBox

app = TaurusApplication(sys.argv)


panel = Qt.QWidget()
layout = Qt.QHBoxLayout()

model = 'sys/tg_test/1/short_scalar_w'

lbl = TaurusLabel()
lbl.model = model

cmb = TaurusValueComboBox()
cmb.model = 'sys/tg_test/1/short_scalar_w'
cmb.addValueNames([('OPTION1', 0), ('OPTION2', 1), ('OPTION3', 2)])
cmb.setAutoApply(True)

layout.addWidget(lbl)
layout.addWidget(cmb)

panel.setLayout(layout)
panel.show()
sys.exit(app.exec_())
```

**Comment by mkomel** [comment]
Ok, I tried this again, now using the newer (release candidate) TangoBox 9.2, and it works fine there.

Previously I was using TangoBox 9.1 (with TaurusQt upgraded to 4.1.1 by:  pip install --upgrade taurus)
Not sure what is the reason for odd behaviour in TangoBox 9.1, one possibilty is due to different PyTango versions (TangoBox 9.1 has PyTango 8.1.2, while TangoBox 9.2 uses a newer version).

Thanks for trying to reproduce this!

**Comment by mkomel** [comment]
Note that I can see this problem when using freshly downloaded TangoBox 9.2 RC10.

With the above sample, when changing an option in the dropdown I can see this output in the console:
```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/taurus/qt/qtgui/base/taurusbase.py", line 1909, in writeValue
    applied = self.safeApplyOperations()
  File "/usr/local/lib/python2.7/dist-packages/taurus/qt/qtgui/base/taurusbase.py", line 1754, in safeApplyOperations
    self.applyPendingOperations(ops)
  File "/usr/local/lib/python2.7/dist-packages/taurus/qt/qtgui/base/taurusbase.py", line 978, in applyPendingOperations
    go()
  File "/usr/local/lib/python2.7/dist-packages/taurus/qt/qtgui/dialog/taurusmessagebox.py", line 193, in wrapped
    msgbox = _ProtectMessageBox(*sys.exc_info())
  File "/usr/local/lib/python2.7/dist-packages/taurus/qt/qtgui/dialog/taurusmessagebox.py", line 166, in _ProtectMessageBox
    err_value=err_value, err_traceback=err_traceback, parent=parent)
  File "/usr/local/lib/python2.7/dist-packages/taurus/qt/qtgui/dialog/taurusmessagebox.py", line 76, in __init__
    self, designMode)
RuntimeError: maximum recursion depth exceeded while calling a Python object
```

**Comment by cpascual** [comment]
Yes, I can confirm it. , So this is clearly a bug. Thanks for reporting!

**Comment by cpascual** [comment]
Curiously enough, the bug does not happen when I am in debug mode in pycharm...

**Comment by cpascual** [comment]
While investigating it, I noticed that the problem is not triggered with all attributes.

In the case of TangoTest, it is triggered with (at least):
- short_scalar_w
- long_scalar_w
- double_scalar_w
- double_scalar

But it is  *not* triggered with (at least):
- short_scalar
- long_scalar

**Comment by cpascual** [comment]
I created a PR with a fix for the issue (#663 ).
@mkomel , can you please confirm that this solves your issue?

**Comment by mkomel** [comment]
Yes, I can confirm that the fix seems to resolve this issue.
